### PR TITLE
plan: remove parents totally

### PR DESCRIPTION
**Pull request by hanfei1991**
remove parents list.

### CODE REVIEW

File: plan/plan.go [hunk 1]
```diff
@@ -340,21 +336,11 @@ func (p *basePlan) Schema() *expression.Schema {
 	return p.schema
 }
 
-// Parents implements Plan Parents interface.
-func (p *basePlan) Parents() []Plan {
```
Comment (zimulala) [inline plan/plan.go]: Can we remove the field of parents?

Comment (hanfei1991) [reply plan/plan.go]: yes, you are right !

**Comment by shenli** [comment]
LGTM

**Comment by hanfei1991** [comment]
After Optimizer
BenchmarkOptimize/select_count(*)_from_t_group_by_e-4              50000             37373 ns/op           26560 B/op        360 allocs/op
BenchmarkOptimize/select_count(*)_from_t_where_e_<=_10_group_by_e-4                20000             96652 ns/op           65201 B/op        841 allocs/op
BenchmarkOptimize/select_count(*)_from_t_where_e_<=_50-4                           20000             97279 ns/op           66017 B/op        847 allocs/op
BenchmarkOptimize/select_count(*)_from_t_where_c_>_'1'_group_by_b-4                10000            118883 ns/op           75697 B/op       1065 allocs/op
BenchmarkOptimize/select_count(*)_from_t_where_e_=_1_group_by_b-4                  10000            113601 ns/op           76465 B/op       1022 allocs/op
BenchmarkOptimize/select_count(*)_from_t_where_e_>_1_group_by_b-4                  10000            117654 ns/op           79121 B/op       1038 allocs/op
BenchmarkOptimize/select_count(e)_from_t_where_t.b_<=_20-4                         10000            123909 ns/op           81608 B/op       1050 allocs/op
BenchmarkOptimize/select_count(e)_from_t_where_t.b_<=_30-4                         10000            121426 ns/op           81606 B/op       1050 allocs/op
BenchmarkOptimize/select_count(e)_from_t_where_t.b_<=_40-4                         10000            119588 ns/op           81608 B/op       1050 allocs/op
BenchmarkOptimize/select_count(e)_from_t_where_t.b_<=_50-4                         10000            124641 ns/op           81606 B/op       1050 allocs/op
BenchmarkOptimize/select_*_from_t_where_t.b_<=_40-4                                30000             58998 ns/op           40662 B/op        492 allocs/op
BenchmarkOptimize/select_*_from_t_where_t.b_<=_50-4                                30000             57768 ns/op           40662 B/op        492 allocs/op
BenchmarkOptimize/select_*_from_t_where_1_and_t.b_<=_50-4                          20000             58840 ns/op           40966 B/op        499 allocs/op
BenchmarkOptimize/select_*_from_t_where_t.b_<=_100_order_by_t.a_limit_1-4          10000            206752 ns/op          153537 B/op       1821 allocs/op
BenchmarkOptimize/select_*_from_t_where_t.b_<=_1_order_by_t.a_limit_10-4           10000            205577 ns/op          152055 B/op       1783 allocs/op
BenchmarkOptimize/select_*_from_t_use_index(b)_where_b_=_1_order_by_a-4            30000             40820 ns/op           27553 B/op        357 allocs/op
BenchmarkOptimize/select_*_from_t_where_d_<_cast('1991-09-05'_as_datetime)-4               20000             66684 ns/op           40704 B/op        520 allocs/op
BenchmarkOptimize/select_*_from_t_where_ts_<_'1991-09-05'-4                                20000             69640 ns/op           40800 B/op        524 allocs/op


Before Optimize

BenchmarkOptimize/select_count(*)_from_t_group_by_e-4              50000             38625 ns/op           27376 B/op        370 allocs/op
BenchmarkOptimize/select_count(*)_from_t_where_e_<=_10_group_by_e-4                10000            100764 ns/op           67393 B/op        855 allocs/op
BenchmarkOptimize/select_count(*)_from_t_where_e_<=_50-4                           10000            100768 ns/op           68465 B/op        861 allocs/op
BenchmarkOptimize/select_count(*)_from_t_where_c_>_'1'_group_by_b-4                10000            124622 ns/op           78065 B/op       1079 allocs/op
BenchmarkOptimize/select_count(*)_from_t_where_e_=_1_group_by_b-4                  10000            116677 ns/op           78897 B/op       1036 allocs/op
BenchmarkOptimize/select_count(*)_from_t_where_e_>_1_group_by_b-4                  10000            121065 ns/op           81553 B/op       1052 allocs/op
BenchmarkOptimize/select_count(e)_from_t_where_t.b_<=_20-4                         10000            125590 ns/op           84153 B/op       1065 allocs/op
BenchmarkOptimize/select_count(e)_from_t_where_t.b_<=_30-4                         10000            123969 ns/op           84153 B/op       1065 allocs/op
BenchmarkOptimize/select_count(e)_from_t_where_t.b_<=_40-4                         10000            124732 ns/op           84153 B/op       1065 allocs/op
BenchmarkOptimize/select_count(e)_from_t_where_t.b_<=_50-4                         10000            126006 ns/op           84154 B/op       1065 allocs/op
BenchmarkOptimize/select_*_from_t_where_t.b_<=_40-4                                30000             60917 ns/op           41520 B/op        502 allocs/op
BenchmarkOptimize/select_*_from_t_where_t.b_<=_50-4                                30000             59513 ns/op           41520 B/op        502 allocs/op
BenchmarkOptimize/select_*_from_t_where_1_and_t.b_<=_50-4                          30000             62181 ns/op           41824 B/op        509 allocs/op
BenchmarkOptimize/select_*_from_t_where_t.b_<=_100_order_by_t.a_limit_1-4          10000            211420 ns/op          156849 B/op       1841 allocs/op
BenchmarkOptimize/select_*_from_t_where_t.b_<=_1_order_by_t.a_limit_10-4           10000            212347 ns/op          155407 B/op       1804 allocs/op
BenchmarkOptimize/select_*_from_t_use_index(b)_where_b_=_1_order_by_a-4            30000             43291 ns/op           28159 B/op        371 allocs/op
BenchmarkOptimize/select_*_from_t_where_d_<_cast('1991-09-05'_as_datetime)-4               20000             75067 ns/op           41584 B/op        529 allocs/op
BenchmarkOptimize/select_*_from_t_where_ts_<_'1991-09-05'-4                                20000             70033 ns/op           41680 B/op        533 allocs/op

**Pull request merged**